### ISSUE TITLE
Config JSON: Add API methods for saving

### ISSUE THREAD

**Issue by ithinkandicode** [GodotModding/godot-mod-loader#132]
Using Config JSON currently has 1 method, which loads configs. But there's currently no way to save data.

It would be good if we could implement this, as it would save mods (or vanilla games) from having to implement it for each game that uses it.

My approach would be to create 2 methods. One saves the entire config, the other saves a single setting:

```gdscript
func save_mod_config_dictionary(data:Dictionary):
    #...

func save_mod_config_setting(key:String, value):
    #...
```

Might be a good idea to return a bool for success/failure (or a status code, if more info can be returned regarding errors).

To help with this, we could also add a generic utility method, simply for saving JSON file data (or maybe saving a string to a file? I'd need to get started on the code to see what's possible/optimal).

**Comment by ithinkandicode** [comment]
The JSON saving utility method could also be used for this issue:

- #109

**Comment by ithinkandicode** [comment]
- Closed by #142
